scripts/boot_patch.sh

```diff
@@ -231,12 +234,17 @@ if [ -f kernel ]; then
   # After:  [mov w2, #-32768]
   ./magiskboot hexpatch kernel 821B8012 E2FF8F12 && PATCHEDKERNEL=true
 
-  # Force kernel to load rootfs for legacy SAR devices
-  # skip_initramfs -> want_initramfs
-  $SYSTEM_ROOT && ./magiskboot hexpatch kernel \
-  736B69705F696E697472616D667300 \
-  77616E745F696E697472616D667300 \
-  && PATCHEDKERNEL=true
+  # Don't patch on devices that use dynamic partitions, as these devices always
+  # use a ramdisk and patching their kernel may break booting on devices that
+  # did not correctly implement androidboot.force_normal_boot.
```

Comment:
I've noticed that your `boot.img` still contains the `force_normal_boot` as a `/proc/cmdline` parameter. Should this be also removed if `getprop ro.boot.dynamic_partitions`  is true?

![screenshot of diff showing vendor vs patched containing `force_normal_boot` ](https://github.com/topjohnwu/Magisk/assets/2230917/7a6db519-b8cb-4e49-b442-068ecbcb395c)

Reply:
I believe you're looking at the wrong file, this PR neither adds nor removes anything from the cmdline and it shouldn't.

(You may be looking at https://github.com/topjohnwu/Magisk/files/11636973/FP3-8901.4.A.0022.0-boot_magisk_patched_modified.img.zip, which was my first (manual) workaround for this issue, that's not what is created by this PR. If you are using this file, you should be using https://github.com/topjohnwu/Magisk/files/11646684/FP3-8901.4.A.0022.0-boot_magisk_patched_skip_initramfs.img.zip instead. As I've explained in https://github.com/topjohnwu/Magisk/issues/6786#issuecomment-1573963698 the former breaks booting into recovery.)

Reply:
Oh, sorry for the confusion then! Thanks for clarifying so quickly.